### DISCUSSION TITLE
What is the best way to add features?

### DISCUSSION THREAD

**Discussion by andersontz** [VSCodium/vscodium#1940] [General]
I'm planning to start a derivative software, still under the MIT license. My software will have another name, another visual identity and new features, exclusive and different from the extensions we currently have. For example, I need to create, first of all, a screen for the user to register directly in my software to use it. Create new main menus, create a different type of code tab, among many other things. My question is, what would be the best practices for executing the features, changing the main code or creating custom extensions for each of them? My intention is to create new features and preferably they are standard, without the user being able to disable them, for example. How does the community create new features for VSCodium?

**Comment by daiyam** [comment]
We don't add new features in VSCodium.

For your purpose, I would look at https://theia-ide.org/ which its purpose is to create your own IDE.
It is based on the same editor as Visual Studio Code and use the same extensions (some might not work).
But you won't be at the whim of MS which can break things in your fork as they want (if you want to keep in sync with them).

**Reply by andersontz** [reply]
But the fact that the vsCodium project is open, MIT, doesn't exactly mean that I can get a copy to change and distribute? I can not do this? Why? I don't intend to keep synchronized with MS, I want to create a different and derived project.

**Reply by andersontz** [reply]
For Example: https://github.com/getcursor/cursor/

This is based on VSCodium, this is what I aim to do.

**Reply by daiyam** [reply]
>  I can not do this? Why?

You can. I do it (https://github.com/zokugun/MrCode). It's how I got involved with VSCodium.

> For Example: https://github.com/getcursor/cursor/
> This is based on VSCodium, this is what I aim to do.

I'm not sure how they do it.
VSCodium use patches to edit Visual Studio Code which isn't the most ideal way to develop new features. 

> I don't intend to keep synchronized with MS, I want to create a different and derived project.

You should simply fork VSCode and take a look at our patches.

Personally, I would use to Thea to create another IDE like VSCode.

**Reply by andersontz** [reply]
My initial intention was to use VSCode, but I was unsure about the licensing, whether it allows it, and in addition, I was also unsure whether I could disable telemetry for my project.

**Reply by daiyam** [reply]
VSCode's code is under MIT but VSCode's app is not.
So you are free to fork it.

**Comment by xiaochao00** [comment]
@andersontz I have the same idea with you, did you found the method how to develop ours software base on vscodium?
